VideoLocker/src/main/java/org/edx/mobile/http/ErrorHandlingCallback.java

```diff
@@ -0,0 +1,266 @@
+package org.edx.mobile.http;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.edx.mobile.util.images.ErrorUtils;
+import org.edx.mobile.view.common.MessageType;
+import org.edx.mobile.view.common.TaskMessageCallback;
+import org.edx.mobile.view.common.TaskProgressCallback;
+
+import java.io.IOException;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import roboguice.RoboGuice;
+
+/**
+ * Generic abstract implementation of Retrofit's
+ * {@link retrofit2.Callback} interface, that takes care of delivering
+ * status and error information to the proper callbacks. It also
+ * provides (and delegates to) a simpler callback interface for
+ * subclasses, stripping out unnecessary parameters, and redirecting
+ * all responses with error codes to the failure callback method (like
+ * it used to be in the implementation in Retrofit 1).
+ *
+ * @param <T> The successful response body type.
+ */
+public abstract class ErrorHandlingCallback<T> implements Callback<T> {
+    /**
+     * The HTTP request type, for determining what kind of error
+     * message to deliver to the callback.
+     */
+    public enum Type {
```

Comment:
The `Task` class should be considered to be deprecated, but OK I'll try to make this shared (and also assign a `@Deprecated` tag on the `Task` class).